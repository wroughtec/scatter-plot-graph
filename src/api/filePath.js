const path = require('path');

const filePath = dataFile => path.join(__dirname, dataFile);

module.exports = filePath;
