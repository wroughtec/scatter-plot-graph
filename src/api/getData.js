const fs = require('fs');

const getData = filePath => {
  if (filePath) {
    return fs.readFileSync(filePath, { encoding: 'utf8' }, error => console.log('>>', error));
  }

  return null;
};

module.exports = getData;
