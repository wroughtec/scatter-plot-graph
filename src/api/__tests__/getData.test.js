import getData from '../getData';
import filePath from '../filePath';

describe('getData', () => {
  const dataFile = '../__mocks__/dataMock.csv';
  it('should convert map to JSON', () => {
    expect(getData(filePath(dataFile))).toMatchSnapshot();
  });

  it('should be null if no path sent', () => {
    expect(getData()).toBeNull();
  });
});
