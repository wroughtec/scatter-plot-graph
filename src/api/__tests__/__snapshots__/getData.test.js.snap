// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData should convert map to JSON 1`] = `
"x,y
18.71230688,11034.12
13.61320583,16340.26
17.58212046,25200.85
0.33767329,503.22
11.34590321,7029.99
65.25484819,38744.4
"
`;
